app/assets/javascripts/discourse/app/components/sidebar/section.js

```diff
@@ -2,41 +2,54 @@ import Component from "@glimmer/component";
 import { tracked } from "@glimmer/tracking";
 import { action } from "@ember/object";
 import { service } from "@ember/service";
+import { bind } from "discourse-common/utils/decorators";
 
 export default class SidebarSection extends Component {
   @service keyValueStore;
+  @service sidebarState;
 
-  @tracked displaySectionContent;
+  @tracked collapsedSections = this.sidebarState.collapsedSections;
   sidebarSectionContentID = `sidebar-section-content-${this.args.sectionName}`;
   collapsedSidebarSectionKey = `sidebar-section-${this.args.sectionName}-collapsed`;
 
-  constructor() {
-    super(...arguments);
-
-    if (this.args.collapsable) {
-      this.displaySectionContent =
+  @bind
+  setExpandedState() {
+    const isCollapsed = (() => {
```

Comment:
Could you move this to a function inside the class also with `@bind` on it? Then you could call `this.isCollapsed` below